content/browser/attribution_reporting/aggregatable_attribution_utils.cc

```diff
@@ -121,6 +129,114 @@ std::vector<AggregatableHistogramContribution> CreateAggregatableHistogram(
   return contributions;
 }
 
+
+// TODO(kelly): Organize in classes
+void AttributionLogicLastTouch(Partition& partition, 
+        base::flat_map<std::string, std::vector<absl::uint128>>& trigger_keypieces_per_source) {
+
+  // for (uint64_t i=attribution_window.epoch_start(); i<=attribution_window.epoch_end(); i++) {
+  //   for (auto& pair : epoch_sources_count[i]) {
+  //     auto& source_key_piece = pair.first;
+  //     auto count = pair.second;
+
+  //     auto it = source_counts.find(source_key_piece);
+  //     if (it == source_counts.end()) {
+  //       source_counts[source_key_piece] = count;
+  //     }
+  //     source_counts[source_key_piece] += count;
+  //   }
+  // }
+}
```

Comment:
So the current prototype only does Uniform? And ARA only does LastTouch?

Reply:
Yes I haven't implemented the last touch yet but should be trivial. 
and yes ARA only does LastTouch. they filter the one and only source that they match much earlier in the code so it was impossible to do uniform. Now we have all relevant sources in the struct at this execution point so we can either do uniform with them or trivially filter them by last-touch